src/test/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializerTest.java

```diff
@@ -135,6 +136,37 @@ public void mustHandleBoolean() throws Exception {
       assertThat(filterClause.comparisonExpressions()).hasSize(1).contains(expectedResult);
     }
 
+    @Test
+    public void mustHandleDate() throws Exception {
+      String json = """
+            {"dateType": {"$date": 1672531200000}}
```

Comment:
I took a look and MongoDB has an [Extended JSON specification](https://www.mongodb.com/docs/manual/reference/mongodb-extended-json/#mongodb-bsontype-Date). It looks like v2 of the spec requires dates be represented as either `{ $date: '2023-04-22T01:13:08.562Z' }` or `{ $date: { $numberLong: '1682126010903' } }` .

`{ $date: 1682126010903 }` is consistent with "shell mode" in v1 of MongoDB's EJSON spec: https://www.mongodb.com/docs/manual/reference/mongodb-extended-json-v1/#date, so this syntax is not quite consistent with modern extended JSON. What do you think about changing this to require `$date: '2023-04-22T01:13:08.562Z'` ISO 8601 date string instead of millis since epoch? That is valid in both EJSON v1 and EJSON v2.

Reply:
Converting between textual format and number is very inefficient so ideally we would use timestamp. So while ISO-8601 String is more readable, it is rather wasteful with regards to CPU costs. :-(

One question is the value for `$numberLong` really String as well, and not JSON number?

Reply:
Number timestamp is fine if you have a strong preference, which it sounds like you do :) No need to abide exactly by the EJSON spec, it's just a convenience.

I suspect why they made the `$numberLong` string change for EJSON v2 is that the JSON spec doesn't explicitly specify the maximum size of JSON numbers, so there may be some JSON parser for some language that doesn't

Reply:
@vkarpov15 Ok that makes sense. And yes, I agree -- it'd be nice to be compliant with something that exists, but we are not bound. One thing we could support if it makes sense to also allow textual values. But let's use timestamp for now.